packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts

```diff
@@ -880,6 +876,82 @@ export class BaseResolversVisitor<
     return `Array<${t}>`;
   }
 
+  protected createResolversUnionTypes(): Record<string, string> {
+    if (!this._hasReferencedResolversUnionTypes) {
+      return {};
+    }
+
+    const allSchemaTypes = this._schema.getTypeMap();
+
+    const unionTypes = Object.entries(allSchemaTypes).reduce((res, [typeName, schemaType]) => {
+      if (isUnionType(schemaType)) {
+        const referencedTypes = schemaType.getTypes().map(unionMemberType => {
+          const isUnionMemberMapped = this.config.mappers[unionMemberType.name];
+
+          // 1. If mapped without plachoder, just use it without doing extra checks
+          if (isUnionMemberMapped && !hasPlaceholder(isUnionMemberMapped.type)) {
+            return isUnionMemberMapped.type;
+          }
+
+          // 2. Work out value for union member type
+          // 2a. By default, use the typescript type
+          let unionMemberValue = this.convertName(unionMemberType.name);
+
+          // 2b. Find fields to Omit if needed.
+          //  - If no field to Omit, "type with maybe Omit" is typescript type i.e. no Omit
+          //  - If there are fields to Omit, "type with maybe Omit"
+          const fields = unionMemberType.getFields();
+          const fieldsToOmit = this._federation
+            .filterFieldNames(Object.keys(fields))
+            .map(fieldName => {
+              const field = fields[fieldName];
+              const baseType = getBaseType(field.type);
+              const isMapped = this.config.mappers[baseType.name];
+              const isUnion = isUnionType(baseType);
+              const shouldMapType = this._shouldMapType[baseType.name];
+
+              if (!isMapped && !isUnion && !shouldMapType) {
+                return null;
+              }
+
+              return {
+                addOptionalSign: !this.config.avoidOptionals && !isNonNullType(field.type),
+                fieldName: field.name,
+                replaceWithType: wrapTypeWithModifiers(this.getTypeToUse(baseType.name), field.type, {
+                  wrapOptional: this.applyMaybe,
+                  wrapArray: this.wrapWithArray,
+                }),
+              };
+            })
```

Comment:
This is exactly the same as the one in `createResolversFields`, will need to abstract to reuse